qiskit_experiments/library/randomized_benchmarking/rb_experiment.py

```diff
@@ -206,284 +233,136 @@ def _sequences_to_circuits(
-            if clbits:
-                creg = ClassicalRegister(clbits)
-                new_cargs = [Clbit(creg, i) for i in range(clbits)]
-                new_circ.add_register(creg)
-
-            for inst, qargs, cargs in circ.data:
-                mapped_cargs = [new_cargs[circ.find_bit(clbit).index] for clbit in cargs]
-                mapped_qargs = [circ.qubits[qargs_map[circ.find_bit(i).index]] for i in qargs]
-                new_circ.data.append((inst, mapped_qargs, mapped_cargs))
-                # Add the calibrations
-                for gate, cals in circ.calibrations.items():
-                    for key, sched in cals.items():
-                        new_circ.add_calibration(gate, qubits=key[0], schedule=sched, params=key[1])
-
-            transpiled.append(new_circ)
-        return transpiled
+        return op.adjoint()
 
     def _transpiled_circuits(self) -> List[QuantumCircuit]:
         """Return a list of experiment circuits, transpiled."""
-        if self.num_qubits in [1, 2]:
-            transpiled = self._layout_for_rb()
-        else:
+        has_custom_transpile_option = (
+            any(
+                opt not in {"basis_gates", "optimization_level"}
+                for opt in vars(self.transpile_options)
+            )
+            or self.transpile_options.get("optimization_level", 0) != 0
+        )
+        if self.num_qubits > 2 or has_custom_transpile_option:
             transpiled = super()._transpiled_circuits()
-        if self.analysis.options.get("gate_error_ratio", None) is None:
+        else:
+            transpiled = [
+                _transpile_clifford_circuit(circ, layout=self.physical_qubits)
+                for circ in self.circuits()
+            ]
+            # Set custom calibrations provided in backend
+            if self.backend:
+                # TODO: Remove V2 restriction after V2 conversion in _set_backend
+                if not isinstance(self.backend, BackendV2):
+                    raise NotImplementedError
+
+                # assert self.num_qubits <= 2
+                qargs_patterns = [self.physical_qubits]  # for self.num_qubits == 1
+                if self.num_qubits == 2:
+                    qargs_patterns = [
+                        (self.physical_qubits[0],),
+                        (self.physical_qubits[1],),
+                        self.physical_qubits,
+                        (self.physical_qubits[1], self.physical_qubits[0]),
+                    ]
+
+                instructions = []  # (op_name, qargs) for each element where qargs means qubit tuple
+                for qargs in qargs_patterns:
+                    for op_name in self.backend.target.operation_names_for_qargs(qargs):
+                        instructions.append((op_name, qargs))
+
+                common_calibrations = defaultdict(dict)
+                for op_name, qargs in instructions:
+                    inst_prop = self.backend.target[op_name][qargs]
```

Comment:
Not sure second dict covers all qubits (user can add custom instruction with limited qubit set). So this is safer
```suggestion
                    inst_prop = self.backend.target[op_name].get(qargs, None)
```

Reply:
Good catch, thanks.